app/services/unified_chat_service.py

```
@@ -609,16 +800,62 @@ async def _check_strategy_access(self, user_id: str) -> Dict[str, Any]:
+                else:
+                    pct = 0.0
+                if pct > largest_position_pct:
+                    largest_position_pct = pct
+                    largest_position_symbol = position.get("symbol")
+
+            max_position_pct_allowed = mode_config.max_position_pct if mode_config else 10.0
+            within_limits = largest_position_pct <= max_position_pct_allowed
+
+            try:
+                portfolio_heat = self.portfolio_risk.position_sizing_engine._calculate_portfolio_heat(portfolio)
+            except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)